src/ReaderThemeSupportFeatures.php

```diff
@@ -0,0 +1,255 @@
+<?php
+/**
+ * Class ReaderThemeSupportFeatures.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use AMP_Options_Manager;
+
+/**
+ * Stores the primary theme's theme support features when Reader template mode is active and then adds the necessary styles to support them.
+ *
+ * Note that this service is not conditional based on whether Reader mode is selected because it needs to run logic
+ * when transitioning from non-reader to Reader mode.
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.1
+ * @internal
+ */
+final class ReaderThemeSupportFeatures implements Service, Registerable {
+
+	/**
+	 * Theme feature slug for editor-color-palette.
+	 *
+	 * @var string
+	 */
+	const FEATURE_EDITOR_COLOR_PALETTE = 'editor-color-palette';
+
+	/**
+	 * Theme feature slug for editor-gradient-presets.
+	 *
+	 * @var string
+	 */
+	const FEATURE_EDITOR_GRADIENT_PRESETS = 'editor-gradient-presets';
+
+	/**
+	 * Theme feature slug for editor-font-sizes.
+	 *
+	 * @var string
+	 */
+	const FEATURE_EDITOR_FONT_SIZES = 'editor-font-sizes';
+
+	/**
+	 * Supported features.
+	 *
+	 * @var string[]
+	 */
+	const SUPPORTED_FEATURES = [
+		self::FEATURE_EDITOR_COLOR_PALETTE,
+		self::FEATURE_EDITOR_GRADIENT_PRESETS,
+		self::FEATURE_EDITOR_FONT_SIZES,
+	];
+
+	/**
+	 * Reader theme loader.
+	 *
+	 * @var ReaderThemeLoader
+	 */
+	private $reader_theme_loader;
+
+	/**
+	 * ReaderThemeLoader constructor.
+	 *
+	 * @param ReaderThemeLoader $reader_theme_loader Reader theme loader.
+	 */
+	public function __construct( ReaderThemeLoader $reader_theme_loader ) {
+		$this->reader_theme_loader = $reader_theme_loader;
+	}
+
+	/**
+	 * Register the service with the system.
+	 *
+	 * @return void
+	 */
+	public function register() {
+		add_filter( 'amp_options_updating', [ $this, 'filter_amp_options_updating' ] );
```

Comment:
Need to make sure this runs when the AMP options are updated with the version bump after upgrading.